docs/guides/caching.md

```diff
@@ -0,0 +1,137 @@
+# Caching
+
+When using the recommended approach for initializing experiences with this SDK, the logic will fetch the experience entry, the data source (bound entries & assets), and used patterns under the hood and return the final experience object.
+
+To inject a manual caching layer in between, the SDK allows you to fetch the content on your own.
+
+## Typical Fetching & Rendering
+
+The simple setup would usually look something like this calling one of the sophisticated fetcher functions `useFetchBySlug`. While the final experience object exposes all fetched content, it doesn't provide a way to inject already fetched content but will fetch everything from the API at once.
+
+```ts
+import { useFetchBySlug } from '@contentful/experiences-sdk-react';
+
+const { experience, error, isLoading } = useFetchBySlug({
+  slug,
+  localeCode: locale,
+  client,
+  experienceTypeId,
+});
+```
+
+## Caching & Injecting Data
+
+To cache the loaded entities and inject them manually, a custom fetcher function needs to be written that uses `fetchExperienceEntry`, `fetchReferencedEntities`, and `createExperience` which are exposed from the SDK as well.
+
+### Fetching Data
+
+```ts
+import {
+  fetchExperienceEntry,
```

Comment:
It doesn't have to be in this PR, but generally, all the functions we export from the SDK and refer to in guides/documentation should have top-notch Doxygen-style comments to guide developers without leaving the IDE. I think this isn't the case for some of these imports right now.

Reply:
Yes, that's a good call-out. Would be great to prepare one or two perfect examples that we aligned on, so it's clear what requirements those should match.
I added docstrings for all exported functions that are used in here, though I'm not sure what level of detail they should have.